hw/ip/otbn/dv/rig/rig/gens/edge_load_store.py

```diff
@@ -0,0 +1,234 @@
+        if len(self.bnlid.operands) != 5:
+            raise RuntimeError('Unexpected number of operands for bn.lid')
+
+        if not (isinstance(self.bnlid.operands[0].op_type, RegOperandType) and
+                (self.bnlid.operands[0].op_type.reg_type == 'gpr' and
+                not self.bnlid.operands[0].op_type.is_dest() and
+                isinstance(self.bnlid.operands[1].op_type, RegOperandType) and
+                self.bnlid.operands[1].op_type.reg_type == 'gpr' and
+                not self.bnlid.operands[1].op_type.is_dest() and
+                isinstance(self.bnlid.operands[2].op_type, ImmOperandType) and
+                self.bnlid.operands[2].op_type.signed and
+                isinstance(self.bnlid.operands[3].op_type, OptionOperandType) and
+                isinstance(self.bnlid.operands[4].op_type, OptionOperandType))):
+            raise RuntimeError('BN.LID instruction from instructions file is'
+                               'not the shape expected by MisalignedBadLoadStore'
+                               'generator.')
+
+        # BN.SID Instruction Checks
+
+        self.bnsid = self._get_named_insn(insns_file, 'bn.sid')
+
+        # bn.sid expects the operands: grs1, grs2, offset, grs1_inc,
+        # grs2_inc
+        if len(self.bnsid.operands) != 5:
+            raise RuntimeError('Unexpected number of operands for bn.sid')
+
+        if not (isinstance(self.bnsid.operands[0].op_type, RegOperandType) and
+                self.bnsid.operands[0].op_type.reg_type == 'gpr' and
+                not self.bnsid.operands[0].op_type.is_dest() and
+                isinstance(self.bnsid.operands[1].op_type, RegOperandType) and
+                self.bnsid.operands[1].op_type.reg_type == 'gpr' and
+                not self.bnsid.operands[1].op_type.is_dest() and
+                isinstance(self.bnsid.operands[2].op_type, ImmOperandType) and
+                self.bnsid.operands[2].op_type.signed and
+                isinstance(self.bnsid.operands[3].op_type, OptionOperandType) and
+                isinstance(self.bnsid.operands[4].op_type, OptionOperandType)):
+            raise RuntimeError('BN.SID instruction from instructions file is'
+                               'not the shape expected by MisalignedBadLoadStore'
+                               'generator.')
+
+        for insn in insns_file.insns:
+            if (insn.mnemonic in ['lw', 'sw', 'bn.lid', 'bn.sid']):
+                weight = cfg.insn_weights.get(insn.mnemonic)
+                if weight > 0:
+                    self.weights.append(weight)
+                    self.insns.append(insn)
+
+        # Check that at least one instruction has a positive weight
+        assert len(self.insns) == len(self.weights)
+        if not self.weights:
+            raise ValueError('Config at {} defines a zero weight '
+                             'for all instructions.'
+                             .format(cfg.path))
```

Comment:
This isn't necessarily an error; it just means the generator should disable itself. See e.g. the branch generator for an example.